src/modules/esl-share/core/esl-share-button.ts

```diff
@@ -34,8 +46,42 @@ export class ESLShareButton extends ESLBaseElement {
   /** Marker of availability of share button */
   @boolAttr() public unavailable: boolean;
 
+  /** @readonly Ready state marker */
+  @boolAttr({readonly: true}) public readonly ready: boolean;
+
+  /** @returns config of button specified by the name attribute */
+  @memoize()
+  public get config(): ESLShareButtonConfig | undefined {
+    return ESLShareConfig.instance.getButton(this.name);
+  }
+
+  /** Gets a property from attribute, or from button config if not set attribute */
+  protected get(prop: 'action' | 'link'): string;
+  protected get(prop: 'additional'): Record<string, any>;
+  protected get(prop: 'action' | 'link' | 'additional'): string | Record<string, any> {
+    return (prop === 'additional')
+      ? Object.keys(this[prop]).length ? this[prop] : this.config?.[prop] || {}
+      : this[prop] || this.config?.[prop] || '';
```

Comment:
Just a suggestion, maybe we can do it more readable. For example:
```suggestion
    if (prop === 'additional') Object.keys(this[prop]).length ? this[prop] : this.config?.[prop] || {};
    else return this[prop] || this.config?.[prop] || '';
```

Reply:
Thanks! Reworked